neps/nep-0508.md

```diff
@@ -22,99 +22,92 @@ While the new approach addresses critical limitations left unsolved in NEP-40 an
 
 ## Motivation
 
-Currently, NEAR protocol has four shards. With more partners onboarding, we started seeing that some shards occasionally become over-crowded. In addition, with state sync and stateless validation, validators will not need to track all shards and validator hardware requirements can be greatly reduced with smaller shard size. 
+Currently, NEAR protocol has four shards. With more partners onboarding, we started seeing that some shards occasionally become over-crowded with respect to total state size and number of transactions. In addition, with state sync and stateless validation, validators will not need to track all shards and validator hardware requirements can be greatly reduced with smaller shard size. With future in-memory tries, it's also important to limit the size of individual shards.
 
 ## Specification
 
 ### High level assumptions
 
-* Flat state is enabled.
-* Shard split boundary is predetermined. In other words, necessity of shard splitting is manually decided.
+* Flat storage is enabled.
+* Shard split boundary is predetermined and hardcoded. In other words, necessity of shard splitting is manually decided.
+* For the time being resharding as an event is only going to happen once but we would still like to have the infrastrcture in place to handle future resharding events with ease.
 * Merkle Patricia Trie is the undelying data structure for the protocol state.
-* Minimal epoch gap between two resharding events is X.
+* Epoch is atleast 6 hrs long for resharding to complete.
 
 ### High level requirements
 
 * Resharding should work even when validators stop tracking all shards.
 * Resharding should work after stateless validation is enabled.
 * Resharding should be fast enough so that both state sync and resharding can happen within one epoch.
-* ~~Resharding should not require additional hardware from nodes.~~
-  * This needs to be assessed during test
-* Resharding should be fault tolerant
-  * A validator should be able to recover in case they go offline during resharding.
-    * For now, our aim is at least allowing a validator to join back after resharding is finished. 
+* Resharding should work efficiently within the limits of the current hardware requirements for nodes.
+* Resharding should be fault tolerant. A validator should be able to recover in case they go offline during resharding.
```

Comment:
Can you add that recovery will be a manual process requiring attention from the node operator?

Reply:
Change the fault tolerant point here